compiler/fir/fir2ir/src/org/jetbrains/kotlin/fir/backend/Fir2IrClassifierStorage.kt

```diff
@@ -113,11 +115,43 @@ class Fir2IrClassifierStorage(
 
     private fun IrClass.declareTypeParameters(klass: FirClass<*>) {
         if (klass is FirRegularClass) {
-            preCacheTypeParameters(klass)
+            if (origin == IrDeclarationOrigin.IR_EXTERNAL_DECLARATION_STUB
```

Comment:
There is external Java stub as well. Should we handle it in a similar way or not necessary?

Reply:
done